apps/tasks/messages.js

```diff
@@ -119,6 +119,71 @@ module.exports = function (grunt) {
     return errors;
   }
 
+  /**
+   * Formats and rewrites i18n string variable values according to locale-specific rules.
+   *
+   * - Localizes Western Arabic numerals to the numeral system of the specified locale, if applicable.
+   *
+   * @param value {string|number} The input value to be formatted.
+   * @param locale {string} The locale identifier used for formatting.
+   * @returns {string|number} The formatted variable value, localized according to the specified locale.
+   */
+  /* eslint-disable prettier/prettier */
+  function localizeInterpolation(value, locale) {
+    try {
+      // Converts the locale to the BCP 47 format, e.g., "en_us" -> "en-US".
+      var lang = locale.replace(/_(\w+)$/, function (match, region) { return '-' + region.toUpperCase(); });
+
+      // Localizes Western Arabic numbers, e.g., "1234.5678" -> "۱٬۲۳۴٫۵۶۷۸" for "fa-IR".
+      value = String(value).replace(/^(\d+(\.\d+)?)$/, function (match, number) {
+        return new Intl.NumberFormat(lang, {
+          useGrouping: false, // Prevents grouping (e.g., 1,000).
+          minimumFractionDigits: number.includes('.') ? number.split('.')[1].length : 0,
+          maximumFractionDigits: 100, // Prevents rounding (100 is max allowed by ECMA-402).
+        }).format(number);
+      });
+    } catch (e) {
+      window.console.error(e);
+    }
+
+    return value;
+  }
+  /* eslint-enable prettier/prettier */
+
+  /**
+   * Overrides the original `MessageFormat.compile` function
+   * to apply the default `fmt.l` formatter to all i18n message variables without a specified format.
+   * @param mf {MessageFormat} The instance of MessageFormat.
+   * @returns {void}
+   */
+  function applyDefaultVariableFormatter(mf) {
+    mf.addFormatters({
+      l: localizeInterpolation, // The default `fmt.l` formatter.
+    });
+
+    const preprocessMessage = msg => {
+      if (typeof msg === 'string') {
+        return msg.replace(/\{((?:[^{}]*|\{[^{}]*\})*)\}/g, (interp, varName) =>
+          varName.includes(',') ? interp : `{${varName}, l}`
```

Comment:
Perhaps...

```suggestion
          // Ignore it if it has an embedded variable of its own
          varName.includes(',')
            ? interp
            : varName.includes('{')
            ? `{${preprocessMessage(varName)}`
            : `{${varName}, l}`
```

Reply:
Yeah, that seems to do the trick:

![image](https://github.com/user-attachments/assets/66cd1c67-f833-4f67-90ec-4351b99e718a)